src/main/java/seedu/address/model/job/JobSalary.java

```diff
@@ -7,16 +7,16 @@
  * Represents a Job's salary in the address book.
  */
 public class JobSalary {
+
     public static final String MESSAGE_CONSTRAINTS =
-            "Job salary should only contain numbers, and it should not be blank";
+            "Job salary should only contain numeric characters, does not start with zero, and it should not be blank";
 
     /*
-     * The first character of the address must not be a whitespace,
-     * otherwise " " (a blank string) becomes a valid input.
+     * The first character of the address must be a digit ranging from 1 to 9. Subsequently, any digit can be inputted.
      */
-    public static final String VALIDATION_REGEX = "[\\p{Digit}][\\p{Digit}]*";
+    public static final String VALIDATION_REGEX = "[1-9][\\p{Digit}]*";
 
-    public final int value;
+    public final String value;
```

Comment:
I do think storing salary as numbers is preferred as one of our potential feature (screen job with salary) need to compare salary, thus we would need to convert this to number eventually. 

And honestly, 2 billions is a pretty reasonable cap for monthly salary RIGHT?

But this works for now, we can keep it as it is.

Reply:
I reverted back to storing as an integer, but with the validation check to see if the input is a valid integer (by directly doing the parseInt test and if there is an exception return False).

<img width="674" alt="image" src="https://github.com/user-attachments/assets/9cecca06-62bd-45f6-80d5-d08a928fb3e9">